pkg/roachpb/data.go

```diff
@@ -1545,6 +1545,18 @@ func (t Transaction) SafeFormat(w redact.SafePrinter, _ rune) {
 	}
 	w.Printf("meta={%s} lock=%t stat=%s rts=%s wto=%t gul=%s",
 		t.TxnMeta, t.IsLocking(), t.Status, t.ReadTimestamp, t.WriteTooOld, t.GlobalUncertaintyLimit)
+	for i, obs := range t.ObservedTimestamps {
```

Comment:
I'm honestly surprised the test impact of this change isn't larger. It would be nice if SafeFormat also passed us the modifiers, because then we could do thinks like only do this for `+v` instead of just `v`.  Anyway, that isn't for this PR.


For this PR, do you think it would make sense to truncate this if there was a very large number of nodes?

Reply:
Good point. I removed this, will open a separate PR.